packages/plugin-types/src/auth_resources.ts

```diff
@@ -54,4 +55,21 @@ export type AuthResources = {
    * L1 Cfn Resources, for when dipping down a level of abstraction is desirable.
    */
   cfnResources: AuthCfnResources;
+  /**
+   * The generated resources for user pool groups
+   */
+  groups: {
+    /**
+     * The group name
+     */
+    groupName: string;
+    /**
+     * The generated CfnUserPoolGroup for this group
+     */
+    cfnUserGroup: CfnUserPoolGroup;
+    /**
+     * The generated Role for this group
+     */
+    role: IRole;
+  }[];
```

Comment:
@josefaidt are you sure you want the override access pattern to be `backend.auth.resources.groups[0].role` rather than `backend.auth.resoruces.groups.admin.role`? I feel pretty strongly that the latter is more readable than wondering what the 0th index of the array is

Reply:
I agree that it is easier to work with an object, and something that was brought up earlier, but landed on this approach since it aligns with the input type.

Ideally you would use `Array.find` and a truthy check/non-null assertion to override groups

```ts
const admins = backend.auth.resources.groups.find(g => g.name === "admins")!
```

Reply:
Chatted offline and we agreed to go with the object access pattern on the resources object

Reply:
Thanks for the feedback; I've updated the PR with the suggested access pattern